### PR TITLE
[Feature] optimize query when only partition columns are involved

### PR DESCRIPTION
**Pull request by dirtysalt**
Fixes #issue

This PR is to optimize the query when only parittion columns are involved. Because we don't need to read any field from file. And to do this:
- reconginze this pattern by rule `PartitionColumnValueOnlyOnScanRule`. the requirements are:
   - only partition columns are output by scan operator
   - agg function is like min/max/count distinct
   - in theory, we support more complex agg function like `min(dt) + max(dt)`. However it's very hard to enumerate all of them. (And we can add more if we like)
- When pattern matched,  mark scan operator as `usePartitionColumnValueOnly = true`
- in BE side, we just output partition columns.

---

The good side of putting this code to BE instead of FE, is the BE have already done a lot of heavy-lift work(like predicate and agg).  

And we support multiple partition columns instead of one. The following query could also use this optimization: (l_shipdate and l_returnflags are both partition columns)
```
select min(l_returnflag), max(l_shipdate) from lineitem_partition 
     where days_add(l_shipdate, ascii(l_returnflag))>= '1992-02-01';
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 26 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|4|7|57.14%|[553, 557, 558]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 155 / 181 (85.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/HudiScanNode.java|0|5|00.00%|[154, 155, 156, 157, 158]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|3|00.00%|[238, 239, 240]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|3|00.00%|[351, 352, 353]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|3|00.00%|[281, 282, 283]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|6|11|54.55%|[926, 1056, 1098, 1143, 1401]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|35|37|94.59%|[219, 228]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MinMaxCountOptOnScanRule.java|44|46|95.65%|[64, 105]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileTableScanNode.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/ScanNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneScanColumnRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalScanOperator.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 40 / 43 (93.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 26 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|4|7|57.14%|[553, 557, 558]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        LogicalAggregationOperator aggregationOperator = (LogicalAggregationOperator) input.getOp();
+        Operator operator = input.getInputs().get(0).getInputs().get(0).getOp();
+        if (!(operator instanceof LogicalScanOperator)) {
+            return false;
+        }
+        LogicalScanOperator scanOperator = (LogicalScanOperator) operator;
+
+        // we can only apply this rule to the queries met all the following conditions:
+        // 1. only contain MIN/MAX/COUNT DISTINCT agg functions
+        // 2. scan operator only output partition columns.(right now we only support 1 column)
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java:58]: check hive table?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        return allValid;
+    }
+
+    private static boolean allPartitionColumnInScanOperator(LogicalScanOperator scanOperator) {
+        int pc = 0;
+        Set<String> partitionColumns = scanOperator.getPartitionColumns();
+        for (ColumnRefOperator c : scanOperator.getOutputColumns()) {
+            if (!partitionColumns.contains(c.getName())) {
+                // materialized column could be taken durding prune stage.
+                if (!scanOperator.getScanOptimzeOption().getCanUseAnyColumn()) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java:84]: select count(1) from tbl where partition_col = 'xxx' will also use this rule to optimize?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java (hunk 1 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java:58]: No need to check. Any table should work.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 40 / 43 (93.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2573, 2574]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 26 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|4|7|57.14%|[584, 588, 589]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2573, 2574]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 26 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|4|7|57.14%|[584, 588, 589]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2573, 2574]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 26 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|4|7|57.14%|[584, 588, 589]|

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2573, 2574]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 36 (08.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|3|17|17.65%|[253, 259, 260, 261, 263, 264, 266, 267, 269, 270, 272, 273, 608, 609]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2573, 2574]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 3 / 36 (08.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|3|17|17.65%|[253, 259, 260, 261, 263, 264, 266, 267, 269, 270, 272, 273, 608, 609]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 46 (80.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2643, 2644]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|2|4|50.00%|[39, 40]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|27|32|84.38%|[46, 73, 96, 97, 98]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 46 (80.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2643, 2644]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|2|4|50.00%|[39, 40]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|27|32|84.38%|[46, 73, 96, 97, 98]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 46 (80.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2643, 2644]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|2|4|50.00%|[39, 40]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|27|32|84.38%|[46, 73, 96, 97, 98]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 46 (80.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2643, 2644]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|2|4|50.00%|[39, 40]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|27|32|84.38%|[46, 73, 96, 97, 98]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[2643, 2644]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PartitionColumnValueOnlyOnScanRule.java|31|32|96.88%|[46]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalScanOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ScanOptimzeOption.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/HdfsScanNode.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 10 / 37 (27.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.cpp|0|17|00.00%|[21, 22, 23, 25, 26, 27, 28, 29, 30, 32, 33, 36, 37, 38, 40, 42, 43]|
|:large_blue_circle:|src/exec/hdfs_scanner_partition.h|0|2|00.00%|[23, 24]|
|:large_blue_circle:|src/connector/hive_connector.cpp|10|18|55.56%|[258, 273, 274, 276, 277, 279, 627, 628]|

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Pull request merged**